### DISCUSSION TITLE
The smoke button counter doesnt work in multi when spectating

### DISCUSSION THREAD

**Discussion by Loreos7** [ppy/osu#20928] [Q&A]
https://user-images.githubusercontent.com/86934170/197802251-46cbb18c-7ee7-44ec-924b-0d01189ec866.mp4

**Comment by goodtrailer** [comment]
The key counter stopped counting because the player pressed smoke during a break, so this is unrelated to multi/spectate. You can actually see that the player in the bottom left presses smoke as well—notably right *after* the break ends—and their key counter increments. This is how all keys/actions work, not sure if smoke should be treated differently because it's a non-gameplay action.

**Reply by peppy** [reply]
It's a bit of a weird edge case. Also the fact that smoke is listed as "B3" is very silly. If it is to be shown there it should be "SMK" or something.

**Reply by goodtrailer** [reply]
Yeah, I think using the underlying enum values is generally silly actually. Mania ends up starting at B10. Catch gets B1/2/3, which simply make no sense in the context of ctb (should be ←/→/DASH or something). Taiko has similar issues of unclarity as well. TBH standard was the only one that got a decent key display, until smoke. I don't see a discussion for any of these, which is surprising.

**Reply by peppy** [reply]
It can probably be resolved with a mapping function or even a simple `DescriptionAttribute` on the buttons. But that's a discussion for another day.
